iocore/aio/AIO.cc

```diff
@@ -379,18 +379,15 @@ aio_queue_req(AIOCallbackInternal *op, int fromAPI = 0)
 static inline int
 cache_op(AIOCallbackInternal *op)
 {
-  bool read = (op->aiocb.aio_lio_opcode == LIO_READ);
+  typedef ssize_t (*p)(int, void*, size_t, __off_t);
+  p action=(op->aiocb.aio_lio_opcode == LIO_READ)?pread:(p)pwrite;
   for (; op; op = (AIOCallbackInternal *)op->then) {
```

Comment:
Seems reasonable enough.  But why is there a (p) in front of pwrite?

Reply:
Because the signatures of `pread` and `pwrite` are slightly different (`void *` vs. `void const*`) and therefore `pwrite` must be cast to the `pread` signature type.  The `(p)` is a C-style cast.

Reply:
Try something like this (double dog dare ya):  https://godbolt.org/z/WGqK1czWK .

Your version of the template function would take ink_aiocb * a and ssize_t res as parameters.